File: public/public-web/src/test/java/ch/difty/scipamato/publ/web/common/PanelTest.java
package ch.difty.scipamato.publ.web.common;

import org.apache.wicket.markup.html.panel.Panel;
import org.junit.jupiter.api.Test;

import ch.difty.scipamato.publ.web.WicketTest;

public abstract class PanelTest<T extends Panel> extends WicketTest {

    @Test
    void assertPanel() {
        getTester().startComponentInPage(makePanel());

        assertSpecificComponents();

        getTester().assertNoErrorMessage();
        getTester().assertNoInfoMessage();
    }

    protected abstract T makePanel();

    protected abstract void assertSpecificComponents();
}

File: public/public-web/src/test/kotlin/ch/difty/scipamato/publ/web/common/PanelTest.kt
package ch.difty.scipamato.publ.web.common

import ch.difty.scipamato.publ.web.WicketTest
import org.apache.wicket.markup.html.panel.Panel
import org.junit.jupiter.api.Test

abstract class PanelTest<T : Panel?> : WicketTest() {

    @Test
    fun assertPanel() {
        tester.startComponentInPage(makePanel())
        assertSpecificComponents()
        tester.assertNoErrorMessage()
        tester.assertNoInfoMessage()
    }

    protected abstract fun makePanel(): T
    protected abstract fun assertSpecificComponents()
}